sci_strategic_plugin/src/sci_strategic_plugin.cpp

```diff
@@ -33,90 +33,206 @@
+  get_parameter<double>("vehicle_accel_limit_multiplier",   config_.vehicle_accel_limit_multiplier);
+  get_parameter<double>("stop_line_buffer",   config_.stop_line_buffer);
+  get_parameter<double>("delta_t",   config_.delta_t);
+  get_parameter<double>("min_gap",   config_.min_gap);
+  get_parameter<double>("reaction_time",   config_.reaction_time);
+  get_parameter<double>("intersection_exit_zone_length",   config_.intersection_exit_zone_length);
+  get_parameter<std::string>("strategic_plugin_name",            config_.strategic_plugin_name);
+  get_parameter<std::string>("lane_following_plugin_name",       config_.lane_following_plugin_name);
+  get_parameter<std::string>("intersection_transit_plugin_name", config_.intersection_transit_plugin_name);
+  get_parameter<std::string>("vehicle_id", config_.vehicle_id);
+  get_parameter<double>("vehicle_length", config_.veh_length);
+  get_parameter<double>("vehicle_deceleration_limit", config_.vehicle_decel_limit);
+  get_parameter<double>("vehicle_acceleration_limit", config_.vehicle_accel_limit);
+  
+   // Register runtime parameter update callback
+  add_on_set_parameters_callback(std::bind(&SCIStrategicPlugin::parameter_update_callback, this, std_ph::_1));
+
+  RCLCPP_INFO_STREAM(rclcpp::get_logger("sci_strategic_plugin"),"Done loading parameters: " << config_);
+
+  // Mobility Operation Subscriber
+  mob_operation_sub_ = create_subscription<carma_v2x_msgs::msg::MobilityOperation>("incoming_mobility_operation", 1, 
+    std::bind(&SCIStrategicPlugin::mobilityOperationCb,this,std_ph::_1));
+  
+  // Current Pose Subscriber
+  current_pose_sub_ = create_subscription<geometry_msgs::msg::PoseStamped>("current_pose", 1, 
+    std::bind(&SCIStrategicPlugin::currentPoseCb,this,std_ph::_1));
+
+  // BSM subscriber
+  bsm_sub_ = create_subscription<carma_v2x_msgs::msg::BSM>("bsm_outbound", 1, 
+    std::bind(&SCIStrategicPlugin::BSMCb,this,std_ph::_1));
+
+  // set world model point form wm listener
+  wm_ = get_world_model();
+  
+  // Setup publishers
+  mobility_operation_pub_ = create_publisher<carma_v2x_msgs::msg::MobilityOperation>("outgoing_mobility_operation", 1);
+
+  // Return success if everthing initialized successfully
+  return CallbackReturn::SUCCESS;
 }
 
+rcl_interfaces::msg::SetParametersResult SCIStrategicPlugin::parameter_update_callback(const std::vector<rclcpp::Parameter> &parameters)
+{
+  auto error_double = update_params<double>({
+    {"vehicle_decel_limit_multiplier", config_.vehicle_decel_limit_multiplier},
+    {"vehicle_accel_limit_multiplier", config_.vehicle_accel_limit_multiplier},
+    {"stop_line_buffer", config_.stop_line_buffer},
+    {"delta_t", config_.delta_t},
+    {"min_gap", config_.min_gap},
+    {"reaction_time", config_.reaction_time},
+    {"intersection_exit_zone_length", config_.intersection_exit_zone_length}
+  }, parameters); // vehicle_acceleration_limit not updated as it's global param
+
+  auto error_string = update_params<std::string>({
```

Comment:
We probably dont want to be able to change the plugins while the node is up? I think we can remove all 3 of them here

Reply:
good point